trieste/models/gpflux/builders.py

```
@@ -0,0 +1,124 @@
+
+from typing import Optional
+
+import gpflow
+import numpy as np
+import tensorflow as tf
+from gpflux.architectures import Config, build_constant_input_dim_deep_gp
+from gpflux.models import DeepGP
+
+from ...data import Dataset
+from ...space import Box, SearchSpace
+from ...types import TensorType
+
+
+def build_vanilla_deep_gp(
+    data: Dataset,
+    num_layers: int = 2,
+    num_inducing_points: Optional[int] = None,
+    search_space: Optional[SearchSpace] = None,
+    inner_layer_sqrt_factor: float = 1e-5,
+    likelihood_variance: float = 1e-2,
+    trainable_likelihood: bool = True,
+) -> DeepGP:
+    """
+    Build a :class:`~gpflux.models.DeepGP` model with sensible initial parameters.
+    We rely on ``build_constant_input_dim_deep_gp`` from :mod:`~gpflux.architectures` to build
+    the model. We found the default configuration used here to work well in most situation,
+    but it should not be taken as a universally good solution.
+
+    :param data: Dataset from the initial design, used to estimate the variance of observations
+        and to provide query points which are used to determine inducing point locations with
+        k-means.
+    :param num_layers: Number of layers in deep GP.
+    :param num_inducing_points: Number of inducing points to use in each layer. If left unset
+        (default) it will be set to the number of query points in ``data``.
+    :param search_space: Search space for performing Bayesian optimization. Used for initialization
+        of inducing locations if ``num_inducing_points`` is larger than the amount of data.
+    :param inner_layer_sqrt_factor: A multiplicative factor used to rescale hidden layers. See
+        :class:`~gpflux.architectures.Config` for details.
+    :param likelihood_variance: Initial noise variance in the likelihood function. See
+        :class:`~gpflux.architectures.Config` for details.
+    :return: A :class:`~gpflux.models.DeepGP` model with sensible default settings.
+    """
```

Should we include a `raise` for invalid values?